unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarRenderer.cs

```diff
@@ -205,84 +204,72 @@ private IEnumerator LoadAvatar()
             int wearableCount = resolvedWearables.Count;
-
             for (int index = 0; index < wearableCount; index++)
             {
-                var wearable = resolvedWearables[index];
-
+                WearableItem wearable = resolvedWearables[index];
                 if (wearable == null)
                     continue;
 
-                if (usedCategories.Contains(wearable.category))
-                {
-                    usedCategories.Remove(wearable.category);
-                }
-
-                if (!wearableControllers.ContainsKey(wearable))
+                unusedCategories.Remove(wearable.category);
+                if (wearableControllers.ContainsKey(wearable))
                 {
-                    AddWearableController(wearable);
+                    UpdateWearableController(wearable);
                 }
                 else
                 {
-                    UpdateWearableController(wearable);
+                    AddWearableController(wearable);
                 }
             }
 
-            foreach (var category in usedCategories)
+            foreach (var category in unusedCategories)
             {
                 switch (category)
                 {
-                    case WearableLiterals.Categories.EYES:
-                        SetDefaultEyes(bodyShapeController.bodyShapeId);
+                    case Categories.EYES:
+                        eyesController = FacialFeatureController.CreateDefaultFacialFeature(bodyShapeController.bodyShapeId, Categories.EYES, eyeMaterial);
                         break;
-                    case WearableLiterals.Categories.MOUTH:
-                        SetDefaultMouth(bodyShapeController.bodyShapeId);
+                    case Categories.MOUTH:
+                        mouthController = FacialFeatureController.CreateDefaultFacialFeature(bodyShapeController.bodyShapeId, Categories.MOUTH, mouthMaterial);
                         break;
-                    case WearableLiterals.Categories.EYEBROWS:
-                        SetDefaultEyebrows(bodyShapeController.bodyShapeId);
+                    case Categories.EYEBROWS:
+                        eyebrowsController = FacialFeatureController.CreateDefaultFacialFeature(bodyShapeController.bodyShapeId, Categories.EYEBROWS, eyebrowMaterial);
                         break;
                 }
             }
 
             CleanUpUnusedItems();
 
-            hiddenList = CreateHiddenList();
-
+            HashSet<string> hiddenList = WearableItem.CompoundHidesList(bodyShapeController.bodyShapeId, resolvedWearables);
             bodyShapeController.SetHiddenList(hiddenList);
             if (!bodyShapeController.isReady)
             {
                 bodyShapeController.Load(transform, OnWearableLoadingSuccess, OnBodyShapeLoadingFail);
             }
 
-            foreach (var kvp in wearableControllers)
+            foreach(WearableController wearable in wearableControllers.Values)
```

Comment:
`Values` doesn't generate a copy? This is the main reason I don't use it more often.

Reply:
No, it points to the original collection:

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.values?view=netcore-3.1
```
The returned Dictionary<TKey,TValue>.ValueCollection is not a static copy; instead, the Dictionary<TKey,TValue>.ValueCollection refers back to the values in the original Dictionary<TKey,TValue>. Therefore, changes to the Dictionary<TKey,TValue> continue to be reflected in the Dictionary<TKey,TValue>.ValueCollection.

Getting the value of this property is an O(1) operation.
```